### ISSUE TITLE
Change input caret to a block?

### ISSUE THREAD

**Issue by spotUP** [kristopolous/BOOTSTRA.386#86]
Oldschool terminals didn't use a line as an input caret but a block, could this be implemented?

**Comment by spotUP** [comment]
I forgot to say, a blinking block, ofcourse. :D

**Comment by kristopolous** [comment]
there's caret-color and that's about it. generally speaking fucking with the baseline ux elements like text input with javascript is far harder that one thinks. i still frequently see bugs on facebook's version ... 

Arguably firefox has the api that would make this actually possible in a bug free way https://developer.mozilla.org/en-US/docs/Web/API/CaretPosition ... but yeah, that's about all there is for now.

**Comment by kristopolous** [comment]
as an aside I have put a blocky cursor in just recently ... https://github.com/kristopolous/BOOTSTRA.386/commit/46d7de0f1a9e82285a4502f19bdce6a09412e8bc around there ... there's not integration, testing, documentation yet but that'll come soon

**Comment by kristopolous** [comment]
yeah it was only half done yesterday ... the vertical problem has been alleviated by a hack to scroll to the character start (essentially by adding negative padding when the scrolling happens, keeping the content aligned) see https://github.com/kristopolous/BOOTSTRA.386/commit/144a490ab1252f54bffbc649d6cb6950a41f9606

This feature fixes the other issue.

The horizontal alignment hasn't been addressed.

**Comment by kristopolous** [comment]
also I'll need to revisit this code and do requestAnimationFrame, which is apparently how this is supposed to be done these days so as to not pester complicated sites with a bunch of redraws.

**Comment by kristopolous** [comment]
I pulled that off here: http://9ol.es/tmp/demo.html but it takes some javascript hackery to make it work.
It looks like there's a pure-css solution these days over here: https://css-tricks.com/almanac/properties/c/caret-shape/ so I'll look into that.
